api/messages.proto

```diff
@@ -210,10 +210,12 @@ message ApplicationUplink {
   repeated RxMetadata rx_metadata = 6;
 
   // Settings for the transmission.
-  TxSettings settings = 7 [(gogoproto.nullable) = false, (validate.rules).message.required = true];
+  TxSettings settings = 7 [(validate.rules).message.required = true];
 
   // Server time when the Network Server received the message.
-  google.protobuf.Timestamp received_at = 8 [(gogoproto.stdtime) = true, (gogoproto.nullable) = false];
+  // This field is not validated in the API since ApplicationUplink is used also in AS packages.
```

Comment:
I would leave this commentary out, as this is an implementation detail. The panics in the code are description enough in my opinion.